cli/internal/cloudcmd/create.go

```diff
@@ -215,14 +220,84 @@ func (c *Creator) createAzure(ctx context.Context, cl terraformClient, config *c
 		return clusterid.File{}, err
 	}
 
+	// Patch the attestation policy to allow the cluster to boot while having secure boot disabled.
+	if err := patcher.Patch(ctx, tfOutput.AttestationURL, constants.EncodedAzureMAAPolicy); err != nil {
+		return clusterid.File{}, err
+	}
+
 	return clusterid.File{
-		CloudProvider: cloudprovider.Azure,
-		IP:            tfOutput.IP,
-		InitSecret:    []byte(tfOutput.Secret),
-		UID:           tfOutput.UID,
+		CloudProvider:  cloudprovider.Azure,
+		IP:             tfOutput.IP,
+		InitSecret:     []byte(tfOutput.Secret),
+		UID:            tfOutput.UID,
+		AttestationURL: tfOutput.AttestationURL,
 	}, nil
 }
 
+// PolicyPatcher interacts with Azure to update the attestation policy.
+type PolicyPatcher interface {
+	Patch(ctx context.Context, attestationURL, policy string) error
+}
+
+// NewPolicyPatcher returns a new PolicyPatcher.
+func NewPolicyPatcher() PolicyPatcher {
+	return policyPatcher{}
+}
+
+type policyPatcher struct{}
+
+// Patch updates the attestation policy to the base64-encoded attestation policy JWT for the given attestation URL.
+// https://learn.microsoft.com/en-us/azure/attestation/author-sign-policy#next-steps
+func (p policyPatcher) Patch(ctx context.Context, attestationURL, policy string) error {
+	// hacky way to update the MAA attestation policy. This should be changed as soon as either the Terraform provider supports it
+	// or the Go SDK gets updated to a recent API version.
+	// https://github.com/hashicorp/terraform-provider-azurerm/issues/20804
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create default Azure credentials: %w", err)
+	}
+	token, err := cred.GetToken(context.Background(), azpolicy.TokenRequestOptions{
+		Scopes: []string{"https://attest.azure.net/.default"},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create default Azure credentials: %w", err)
+	}
+
+	client := attestation.NewPolicyClient()
+
+	// azureGuest is the id for the "Azure VM" attestation type. Other types are documented here:
+	// https://learn.microsoft.com/en-us/rest/api/attestation/policy/set
+	req, err := client.SetPreparer(context.Background(), attestationURL, "azureGuest", p.encodeAttestationPolicy(constants.EncodedAzureMAAPolicy))
```

Comment:
```suggestion
	req, err := client.SetPreparer(ctx, attestationURL, "azureGuest", p.encodeAttestationPolicy(constants.EncodedAzureMAAPolicy))
```

Reply:
Looking at this, is it correct that we use the background context and not the (passed down) command context at the rollbacker? If so, can I add this to the godoc of the `rollback` method so that nobody ends up using the wrong context by accident.
https://github.com/edgelesssys/constellation/blob/5355e1cc9c963f7749beb571a60bbd8860fc50d6/cli/internal/cloudcmd/create.go#L217

Reply:
Yes, this is correct because if the context is canceled, the rollback should still be performed. @katexochen, I wonder whether we should remove the context parameter from rollbackOnError and create the background context in the func to prevent wrong use of the func.

Reply:
Yes, good idea!